cluster-autoscaler/cloudprovider/gce/templates.go

```diff
@@ -228,20 +261,63 @@ func parseCustomMachineType(machineType string) (cpu, mem int64, err error) {
 	return
 }
 
+func parseKubeReserved(kubeReserved string) (apiv1.ResourceList, error) {
+	resourcesMap, err := parseLabelsToMap([]string{kubeReserved})
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract kube-reserved from kube-env: %q", err)
+	}
+	reservedResources := apiv1.ResourceList{}
+	for name, quantity := range resourcesMap {
+		switch apiv1.ResourceName(name) {
+		case apiv1.ResourceCPU, apiv1.ResourceMemory, apiv1.ResourceEphemeralStorage:
+			if q, err := resource.ParseQuantity(quantity); err == nil && q.Sign() >= 0 {
```

Comment:
what if error != nil?

Reply:
We just omit the resource which will be treated as reserved=0. Would you like me to log a warning?